internal/k3s/install.go

```diff
@@ -0,0 +1,221 @@
+package k3s
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/weka/gohomecli/internal/bundle"
+)
+
+const k3sImagesPath = "/var/lib/rancher/k3s/agent/images/"
+const defaultLocalStoragePath = "/opt/local-path-provisioner"
+
+var (
+	ErrExists = errors.New("k3s already installed")
+)
+
+var (
+	k3SInstallDir string
+	k3sBinary     string
+
+	k3sBundleRegexp = regexp.MustCompile(`k3s-(.*)\.(tar(\.gz)?)|(tgz)`)
+)
+
+func init() {
+	// by default install to current directory
+	exe, _ := os.Executable()
+	k3SInstallDir = path.Dir(exe)
+	k3sBinary = path.Join(k3SInstallDir, "k3s")
+}
+
+type InstallConfig struct {
+	Iface      string   // interface for k3s network to work on, required
+	BundlePath string   // path to bundle with k3s and images
+	Hostname   string   // host name of the cluster, optional, default from env
+	NodeIP     []string // list of node ip addresses, optional
+	ExternalIP []string // list of external ip addresses, optional
+	dnsFixed   bool
+}
+
+func Install(ctx context.Context, c InstallConfig) error {
+	if hasK3S() {
+		return ErrExists
+	}
+
+	if err := validateNetwork(c.Iface, &c.NodeIP); err != nil {
+		return err
+	}
+
+	fixed, err := dnsFixed()
+	if err != nil {
+		return err
+	}
+	c.dnsFixed = fixed
```

Comment:
why do you need this information inside config?

Reply:
because we need to pass additional parameter to k3s install if it's has workaround

Reply:
let's make it more readable